src/vuln_analysis/utils/chain_of_calls_retriever.py

```diff
@@ -238,6 +238,25 @@ def __find_caller_function(self, document_function: Document, function_package:
         # If didn't find a matching caller function document, returns None.
         return None
 
+    def _is_doc_excluded(self, doc: Document, exclusions: list[Document]) -> bool:
+        """
+        Checks if a document is in the exclusions list based on its
+        function name, function body and source metadata.
+        """
+        doc_function_name = self.language_parser.get_function_name(doc)
+        doc_function_content = doc.page_content
+        doc_source = doc.metadata.get('source')
+
+        for exclusion_doc in exclusions:
+            exclusion_function_name = self.language_parser.get_function_name(exclusion_doc)
+            exclusion_function_content = exclusion_doc.page_content
+            exclusion_source = exclusion_doc.metadata.get('source')
```

Comment:
@TamarW0 Better Strip all results to remove leading and trailing spaces that might be  there